squad/core/management/commands/fix_squadplugin_data.py

```diff
@@ -1,52 +1,32 @@
-import logging
-
 from django.db.models import Q
 from django.core.management.base import BaseCommand
-
+from django.db import IntegrityError
 from squad.core.models import Test, SuiteMetadata
-from squad.core.utils import split_list
-
-
-logger = logging.getLogger()
 
 
 class Command(BaseCommand):
 
-    help = """fix suite in SuiteMetadata for cts/vts tests"""
+    help = """fix suite in SuiteMetadata for buggy_metadata inserted from tradefed plugin bug"""
 
     def handle(self, *args, **options):
-        android_suites = ['cts', 'vts']
-        condition = Q(suite__icontains=android_suites[0])
-        condition |= Q(suite__icontains=android_suites[1])
-        cts_and_vts_metadata = SuiteMetadata.objects.filter(condition, kind='test').order_by('-id')
-        buggy_metadata = []
-        for s in cts_and_vts_metadata:
-            print("Processing SuiteMetedata with suite %s and test %s" % (s.suite, s.name), flush=True)
+        buggy_patterns = [r'^armeabi', r'^arm64']
+        condition = Q(suite__regex=buggy_patterns[0])
+        condition |= Q(suite__regex=buggy_patterns[1])
+        buggy_metadata = SuiteMetadata.objects.filter(condition, kind='test')
+        total = len(buggy_metadata)
+        counter = 0
+        for s in buggy_metadata:
             first_test = Test.objects.filter(metadata=s).first()
```

Comment:
I think you can avoid `first_test = Test.objects.filter(metadata=s).first()` by annotating it to the initial SuiteMetadata query. Something like that:

```python3
from django.db.models import OuterRef, Subquery, F
first_test = Test.objects.filter(metadata=OuterRef('pk')).annotate(suite_slug=F('suite__slug'))
buggy_metadata = SuiteMetadata.objects.annotate(first_test_slug=Subquery(first_test.values('suite_slug')[:1]))
```